library/core/src/result.rs

```diff
@@ -851,7 +880,16 @@ impl<T, E> Result<T, E> {
     /// ```
     #[inline]
     #[stable(feature = "result_map_or_else", since = "1.41.0")]
-    pub fn map_or_else<U, D: FnOnce(E) -> U, F: FnOnce(T) -> U>(self, default: D, f: F) -> U {
+    #[rustc_const_unstable(feature = "const_result_callback_methods", issue = "67792")]
+    pub const fn map_or_else<
+        U,
+        D: ~const FnOnce(E) -> U + ~const Destruct,
+        F: ~const FnOnce(T) -> U + ~const Destruct,
+    >(
+        self,
+        default: D,
+        f: F,
+    ) -> U {
```

Comment:
Same, move the bounds to `where` to avoid this wrapping. `and_then` and `or_else` are similar.